persistence/src/test/scala/hmda/persistence/processing/LocalHmdaEventProcessorSpec.scala

```diff
@@ -113,17 +113,14 @@ class LocalHmdaEventProcessorSpec extends ActorSpec {
     }
   }
 
-  def checkSubmissionStatus(status: SubmissionStatus): Assertion = {
-    //TODO: find a way to avoid having to sleep here
-    Thread.sleep(500)
+  def checkSubmissionStatus(status: SubmissionStatus): Future[Future[Assertion]] = {
     val fSubmissions = (supervisor ? FindSubmissions(SubmissionPersistence.name, submissionId.institutionId, submissionId.period)).mapTo[ActorRef]
-    val fSubmissionStatus = for {
-      a <- fSubmissions
-      b <- (a ? GetState).mapTo[Seq[Submission]]
-    } yield b.head.submissionStatus
-
-    val submissionStatus = Await.result(fSubmissionStatus, duration)
-    submissionStatus mustBe status
+    fSubmissions.map { subActor =>
+      val fSubmissionSeq = (subActor ? GetState).mapTo[Seq[Submission]]
+      fSubmissionSeq.map { subList =>
+        subList.head.submissionStatus mustBe status
```

Comment:
Having this assertion happen async appears not to work, unfortunately. When the assertion fails, it doesn't show a failure in the test results.